worlds/mmbn3/__init__.py

```diff
@@ -172,19 +178,35 @@ def set_rules(self) -> None:
 
         # Set WWW ID requirements
         def has_www_id(state): return state.has(ItemName.WWW_ID, self.player)
-        self.multiworld.get_location(LocationName.ACDC_1_PMD, self.player).access_rule = has_www_id
-        self.multiworld.get_location(LocationName.SciLab_1_WWW_BMD, self.player).access_rule = has_www_id
-        self.multiworld.get_location(LocationName.Yoka_1_WWW_BMD, self.player).access_rule = has_www_id
-        self.multiworld.get_location(LocationName.Undernet_1_WWW_BMD, self.player).access_rule = has_www_id
+        add_rule(self.multiworld.get_location(LocationName.ACDC_1_PMD, self.player), has_www_id)
+        add_rule(self.multiworld.get_location(LocationName.SciLab_1_WWW_BMD, self.player), has_www_id)
+        add_rule(self.multiworld.get_location(LocationName.Yoka_1_WWW_BMD, self.player), has_www_id)
+        add_rule(self.multiworld.get_location(LocationName.Undernet_1_WWW_BMD, self.player), has_www_id)
 
         # Set Press Program requirements
         def has_press(state): return state.has(ItemName.Press, self.player)
-        self.multiworld.get_location(LocationName.Yoka_1_PMD, self.player).access_rule = has_press
-        self.multiworld.get_location(LocationName.Yoka_2_Upper_BMD, self.player).access_rule = has_press
-        self.multiworld.get_location(LocationName.Beach_2_East_BMD, self.player).access_rule = has_press
-        self.multiworld.get_location(LocationName.Hades_South_BMD, self.player).access_rule = has_press
-        self.multiworld.get_location(LocationName.Secret_3_BugFrag_BMD, self.player).access_rule = has_press
-        self.multiworld.get_location(LocationName.Secret_3_Island_BMD, self.player).access_rule = has_press
+        add_rule(self.multiworld.get_location(LocationName.Yoka_1_PMD, self.player), has_press)
+        add_rule(self.multiworld.get_location(LocationName.Yoka_2_Upper_BMD, self.player), has_press)
+        add_rule(self.multiworld.get_location(LocationName.Beach_2_East_BMD, self.player), has_press)
+        add_rule(self.multiworld.get_location(LocationName.Hades_South_BMD, self.player), has_press)
+        add_rule(self.multiworld.get_location(LocationName.Secret_3_BugFrag_BMD, self.player), has_press)
+        add_rule(self.multiworld.get_location(LocationName.Secret_3_Island_BMD, self.player), has_press)
+
+        # Set Purple Mystery Data Unlocker access
+        def can_unlock(state): return state.can_reach(RegionName.SciLab_Overworld, "Region", self.player) or \
```

Comment:
This is a better method (applies to some other changes as well)
```suggestion
        def can_unlock(state): return state.can_reach_region(RegionName.SciLab_Overworld, self.player) or \
```